api/app/Models/PoolCandidate.php

```diff
@@ -273,6 +273,43 @@ public function educationRequirementExperiences(): BelongsToMany
             ->withTimestamps();
     }
 
+    public function getCategoryAttribute()
+    {
+        $category = null;
+
+        switch ($this->user->priority_weight) {
+            case 10:
+                if ($this->user->has_priority_entitlement && $this->priority_verification !== ClaimVerificationResult::REJECTED->name) {
+                    $category = PriorityWeight::PRIORITY_ENTITLEMENT;
+                    break;
+                }
+
+            case 20:
+                if ($this->user->armed_forces_status == ArmedForcesStatus::VETERAN->name && $this->veteran_verification !== ClaimVerificationResult::REJECTED->name) {
+                    $category = PriorityWeight::VETERAN;
+                    break;
+                }
+
+            case 30:
+                $category = PriorityWeight::CITIZEN_OR_PERMANENT_RESIDENT;
+                break;
+
+            case 40:
+                $category = PriorityWeight::OTHER;
+                break;
+
+            default:
+                return null;
+
+        }
+
+        return [
+            'weight' => $category->weight($category->name),
```

Comment:
Both return statements are technically correct, but you need to ensure that $category is not null before accessing its properties in the final return.

Reply:
Category will always set though unless I am missing something? In the switch statement we set category and in the default case we simply return `null`.

https://github.com/GCTC-NTGC/gc-digital-talent/blob/1ccafa91f174e80a3cc0e2e56eb64266bd6fbe67/api/app/Models/PoolCandidate.php#L301-L302

So, once we get to this line, `$category` is set since otherwise it would have fall through to the default case and been returned before reaching this line.